modules/analysis/src/service/load.rs

```diff
@@ -237,9 +248,59 @@ impl AnalysisService {
                 .distinct()
                 .into_query(),
             GraphQuery::Query(query) => sbom_node::Entity::find()
-                .filtering(query.clone())?
+                .join(JoinType::Join, sbom_node::Relation::Package.def())
+                .join(JoinType::LeftJoin, sbom_package::Relation::Purl.def())
+                .join(JoinType::LeftJoin, sbom_package::Relation::Cpe.def())
+                .join(
+                    JoinType::LeftJoin,
+                    sbom_package_cpe_ref::Relation::Cpe.def(),
+                )
                 .select_only()
                 .column(sbom_node::Column::SbomId)
+                .filtering_with(
+                    query.clone(),
+                    sbom_node::Entity
+                        .columns()
+                        .add_columns(cpe::Entity.columns())
+                        .translator(|f, op, v| {
+                            match (f, op, OwnedUri::from_str(v)) {
+                                ("cpe", "=" | "~", Ok(cpe)) => {
+                                    // We break out cpe into its constituent columns in CPE table
+                                    let q = match (cpe.part(), cpe.language()) {
+                                        (CpeType::Any, Language::Any) => String::new(),
+                                        (CpeType::Any, l) => format!("language={l}"),
+                                        (p, Language::Any) => format!("part={p}"),
+                                        (p, l) => format!("part={p}&language={l}"),
+                                    };
+                                    let translated = [
+                                        ("vendor", cpe.vendor()),
+                                        ("product", cpe.product()),
+                                        ("version", cpe.version()),
+                                        ("update", cpe.update()),
+                                        ("edition", cpe.edition()),
+                                    ]
+                                    .iter()
+                                    .fold(q, |acc, (k, v)| match v {
+                                        Component::Value(s) => format!("{acc}&{k}={s}|*"),
+                                        _ => acc,
+                                    });
+                                    Some(translated)
+                                }
+                                ("cpe", "~", Err(_)) => Some(v.into()),
+                                ("cpe", _, Err(e)) => Some(e.to_string()),
+                                ("cpe", _, _) => Some("illegal operation for cpe".into()),
+                                _ => None,
+                            }
+                        })
+                        .add_expr(
+                            "purl",
+                            SimpleExpr::FunctionCall(
+                                Func::cust("get_purl".into_identity())
```

Comment:
FWIW work @ctron has recently done probably means we no longer need to use **get_purl**

Reply:
Oh? I didn't see any related changes beneath `migrations/` -- are we no longer storing purl's in a json blob?

Reply:
we are, but I think the full serialisation is byproduct of the model now

Reply:
lets keep this how you have it for now